homeassistant/components/auth/login_flow.py

```diff
@@ -138,18 +140,61 @@ def _prepare_result_json(result):
     return data
 
 
-class LoginFlowIndexView(HomeAssistantView):
-    """View to create a config flow."""
+class LoginFlowBaseView(HomeAssistantView):
+    """Base class for the login views."""
 
-    url = "/auth/login_flow"
-    name = "api:auth:login_flow"
     requires_auth = False
 
     def __init__(self, flow_mgr, store_result):
         """Initialize the flow manager index view."""
         self._flow_mgr = flow_mgr
         self._store_result = store_result
 
+    async def _async_flow_result_to_response(self, request, client_id, result):
+        """Convert the flow result to a response."""
+        if result["type"] != data_entry_flow.RESULT_TYPE_CREATE_ENTRY:
+            if result["type"] == data_entry_flow.RESULT_TYPE_FORM:
+                # @log_invalid_auth does not work here since it returns HTTP 200
+                # need manually log failed login attempts
+                if result.get("errors", {}).get("base") in (
+                    "invalid_auth",
+                    "invalid_code",
+                ):
+                    await process_wrong_login(request)
+            return self.json(_prepare_result_json(result))
+
+        result.pop("data")
+
+        hass = request.app["hass"]
+        result_obj = result.pop("result")
+
+        user = None
+
+        if isinstance(result_obj, User):
+            user = result_obj
```

Comment:
Let's cover this case in tests

Reply:
I've instead opened https://github.com/home-assistant/core/pull/60468 to remove this case. Let's merge that one first and I will rebase this PR.